src/test/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/RoleServiceTest.kt

```diff
@@ -38,6 +40,89 @@ class RoleServiceTest {
     System.setProperty("http.keepAlive", "false")
   }
 
+  @Nested
+  inner class CreateRoles {
+    @Test
+    fun `create role`() {
+      val createRole = CreateRole(
+        roleCode = "ROLE",
+        roleName = "Role Name",
+        roleDescription = "Role description",
+        adminType = mutableSetOf(AdminType.EXT_ADM)
+      )
+      whenever(roleRepository.findByRoleCode(anyString())).thenReturn(null)
+      whenever(authenticationFacade.currentUsername).thenReturn("user")
+
+      roleService.createRole(createRole)
+      val authority = Authority("ROLE", " Role Name", "Role description", mutableListOf(AdminType.EXT_ADM))
+      verify(roleRepository).findByRoleCode("ROLE")
+      verify(roleRepository).save(authority)
+      verify(telemetryClient).trackEvent(
+        "RoleCreateSuccess",
+        mapOf(
+          "username" to "user",
+          "roleCode" to "ROLE",
+          "roleName" to "Role Name",
+          "roleDescription" to "Role description",
+          "adminType" to "[EXT_ADM]"
+        ),
+        null
+      )
+    }
+
+    @Test
+    fun `create role - having adminType DPS_LSA will auto add DPS_ADM`() {
+      val createRole = CreateRole(
+        roleCode = "ROLE",
+        roleName = "Role Name",
+        roleDescription = "Role description",
+        adminType = mutableSetOf(AdminType.DPS_LSA)
+      )
+      whenever(roleRepository.findByRoleCode(anyString())).thenReturn(null)
+      whenever(authenticationFacade.currentUsername).thenReturn("user")
+
+      roleService.createRole(createRole)
+      val authority =
+        Authority("ROLE", " Role Name", "Role description", mutableListOf(AdminType.DPS_LSA, AdminType.DPS_ADM))
+      verify(roleRepository).findByRoleCode("ROLE")
+      verify(roleRepository).save(authority)
+      verify(telemetryClient).trackEvent(
+        "RoleCreateSuccess",
+        mapOf(
+          "username" to "user",
+          "roleCode" to "ROLE",
+          "roleName" to "Role Name",
+          "roleDescription" to "Role description",
+          "adminType" to "[DPS_LSA, DPS_ADM]"
+        ),
+        null
+      )
+    }
+
+    @Test
+    fun `Create role exists`() {
+      val createRole = CreateRole(
+        roleCode = "NEW_ROLE",
+        roleName = "Role Name",
+        roleDescription = "Role description",
+        adminType = mutableSetOf(AdminType.DPS_LSA)
+      )
+      whenever(roleRepository.findByRoleCode(anyString())).thenReturn(
+        Authority(
+          roleCode = "NEW_ROLE",
+          roleName = "Role Name",
+          roleDescription = "Role description",
+          adminType = mutableListOf(AdminType.DPS_LSA)
+        )
+      )
+
+      Assertions.assertThatThrownBy {
+        roleService.createRole(createRole)
+      }.isInstanceOf(RoleService.RoleExistsException::class.java)
```

Comment:
Can prob just import RoleService.RoleExistsException - to make it neater

Reply:
Done